mne/preprocessing/maxwell.py

```diff
@@ -41,6 +41,136 @@
+        them unchanged. Note that ``'from_raw'`` can only be combined with
+        ``meas_date='from_raw'``.
+    %(verbose)s
+
+    Returns
+    -------
+    raw_er_prepared : instance of Raw
+        A copy of the passed empty-room recording, ready for Maxwell filtering.
+
+    Notes
+    -----
+    This function will:
+
+    * Compile the list of bad channels according to the ``bads`` parameter.
+    * Inject the device-to-head transformation matrix from the experimental
+      recording into the empty-room recording.
+    * Set the following properties of the empty-room recording to match the
+      experimental recording:
+
+      * Montage
+      * ``raw.first_time`` and ``raw.first_samp``
+
+    * Adjust the measurement date according to the ``meas_date`` parameter.
+    * Adjust annotations according to the ``annotations`` parameter.
+
+    .. versionadded:: 1.1
+    """  # noqa: E501
+    _validate_type(item=raw_er, types=BaseRaw, item_name='raw_er')
+    _validate_type(item=raw, types=BaseRaw, item_name='raw')
+    _validate_type(item=bads, types=(list, str), item_name='bads')
+    if isinstance(bads, str):
+        _check_option(
+            parameter='bads', value=bads,
+            allowed_values=['from_raw', 'union', 'keep']
+        )
+    _validate_type(item=meas_date, types=str, item_name='meas_date')
+    _check_option(
+        parameter='meas_date', value=meas_date,
+        allowed_values=['from_raw', 'keep']
+    )
+    _validate_type(item=annotations, types=str, item_name='annotations')
+    _check_option(
+        parameter='annotations', value=annotations,
+        allowed_values=['from_raw', 'keep']
+    )
+
+    if annotations == 'keep' and meas_date != 'keep':
+        raise ValueError('annotations="keep" requires meas_date="keep"')
+    elif annotations == 'from_raw' and meas_date == 'keep':
+        raise ValueError(
+            'annotations="from_raw" requires meas_date="from_raw"'
+        )
+
+    raw_er_prepared = raw_er.copy()
+    del raw_er  # just to be sure
+
+    # handle bads; only keep MEG channels
+
+    if bads == 'from_raw':
+        bads = raw.info['bads']
+    elif bads == 'union':
+        bads = sorted(
+            set(raw.info['bads'] + raw_er_prepared.info['bads'])
+        )
+    elif bads == 'keep':
+        bads = raw_er_prepared.info['bads']
+
+    bads = [ch_name for ch_name in bads
+            if ch_name.startswith('MEG')]
+    raw_er_prepared.info['bads'] = bads
+
+    # handle dev_head_t
+    raw_er_prepared.info['dev_head_t'] = raw.info['dev_head_t']
+
+    # handle montage
+    montage = raw.get_montage()
+    raw_er_prepared.set_montage(montage)
+
+    # handle first_samp
+    raw_er_prepared._cropped_samp = raw._cropped_samp
+
+    # handle meas_date
+    if meas_date == 'from_raw':
+        raw_er_prepared.set_meas_date(raw.info['meas_date'])
+
+    # handle annotations
+    if annotations == 'from_raw':
+        raw_er_prepared.set_annotations(raw.annotations)
```

Comment:
As implemented the logic here is incomplete. If you *dont* set meas_date of `raw_er`, you need to do some gymnastics to `raw.annotations` to get them to appear at the correct time intervals in `raw_er`

Reply:
Currently this behavior is not allowed, see the section at the beginning of the function where we check the parameter values

Reply:
Ahh I didn't notice that. If it's the case that `meas_date` and `annotations` must always match, then there is no reason for two parameters (there is only one degree of freedom here, not two). I'm not sure why they have to match in principle, though...